rosapi/src/rosapi/params.py

```diff
@@ -55,48 +76,139 @@ def set_param(name, value, params_glob):
         d = loads(value)
     except ValueError:
         raise Exception("Due to the type flexibility of the ROS parameter server, the value argument to set_param must be a JSON-formatted string.")
+
+    node_name = get_absolute_node_name(node_name)
     with param_server_lock:
-        rospy.set_param(name, d)
-    
-    
-def get_param(name, default, params_glob):
+        _set_param(node_name, name, value)
```

Comment:
@jubeira Sorry to dig up this old PR, but I'm having some trouble with a client that calls on this API and I think I've isolated the issue to this spot. It seems as though when `value` is a JSON-encoded string, the later call to `get_parameter_value()` sets the `ParameterValue.string_value` attribute with double-quotes, meaning it's impossible to set a string parameter via this method without double quotes being applied.

If the client sends the `value` without JSON encoding, the above `ValueError` is thrown and `set_param()` fails altogether.

I think inside the `try` block, there could be a line `value = d if isinstance(d, str) else value`.

This would ensure that a valid string object is later passed to `get_parameter_value()` rather than a JSON-encoded one, but otherwise behaviour would stay the same. Would this potentially work?

Reply:
Hi @travipross!
Honestly it's been a while so I'm not sure what the actual consequences of that change could be. Your proposal makes sense but I can't be 100% sure without trying the code.
I'd suggest you to fork the repo, apply the change and check if it works for you. If it does, please feel free to contribute the change with a PR explaining what you did.
Thanks!

Reply:
Thanks, opened one here: #521